src/abstractdataframe/selection.jl

```diff
@@ -483,6 +509,9 @@ function _select(df::AbstractDataFrame, normalized_cs, copycols::Bool)
             transformed_cols[newname] = nc
         end
     end
+    # we allow resizing newdf only if up to some point only scalars were put
```

Comment:
I think that this logic would actually supercede all of `combine`'s features. 

Eventually, we want `select` to work on a grouped data frame, performing each operation by group in order to allow `select(gdf, :, :x1 => (x -> x .- mean(x)))`. But with this automatic collapsing with the following call

```
select(gdf, :x1 => mean, :x1 => var)
```

we would get the exact same behavior as `combine`. 

This would be an interesting direction. I was expecting always `spread`ing to have the same number of rows as `nrow` df. 

note that the above ambiguity only applies when you aren't keeping any of the old columns.

Reply:
> Eventually, we want `select` to work on a grouped data frame

Yes - and we want then it to behave exactly like `combine` except that it will return a `GroupedDataFrame` (and not a `DataFrame` like combine)

> I was expecting always spreading to have the same number of rows as `nrow(df)`.

This would conflict with what `combine` does now. We can discuss if `1` row or `nrow(df)` is better in this case, but at least in `combine` we produce `1` row (which is intuitive at least for me).

> note that the above ambiguity only applies when you aren't keeping any of the old columns.

Exactly - this is what I handle in the code.

Reply:
And what is your opinion about spreading to 0 rows - should we allow such a spread or throw an error (current behavior)?

Reply:
> And what is your opinion about spreading to 0 rows - should we allow such a spread or throw an error (current behavior)?

So the current behavior is 

```
julia> select(df, :x1 => (x -> []))
0×1 DataFrame

```

and 

```
julia> select(df,:, :x1 => (x -> []))
ERROR: ArgumentError: New columns must have the same length as old columns
```

Versus `Ref`

julia> select(df,:x1 => (x -> Ref([])))
1×1 DataFrame
│ Row │ x1_function            │
│     │ Array{Any,1}           │
├─────┼────────────────────────┤
│ 1   │ 0-element Array{Any,1} │

julia> select(df,:, :x1 => (x -> Ref([])))
5×4 DataFrame
│ Row │ x1       │ x2         │ a     │ x1_function            │
│     │ Float64  │ Float64    │ Int64 │ Array{Any,1}           │
├─────┼──────────┼────────────┼───────┼────────────────────────┤
│ 1   │ 0.435143 │ 0.849017   │ 1     │ 0-element Array{Any,1} │
│ 2   │ 0.943903 │ 0.00494995 │ 1     │ 0-element Array{Any,1} │
│ 3   │ 0.351229 │ 0.898012   │ 2     │ 0-element Array{Any,1} │
│ 4   │ 0.689727 │ 0.65101    │ 2     │ 0-element Array{Any,1} │
│ 5   │ 0.866909 │ 0.343819   │ 2     │ 0-element Array{Any,1} │
```

If this is what you mean, I think this is good behavior. 

Sorry about not realizing `transform` would preserve grouping. This is good, as long as `combine(gdf, :, :x1 => mean)`  works and "spreads" everything.

Reply:
What I mean with 0-length broadcasting is that:
```
julia> select(df, :, [] => (() -> 9) => :x)
ERROR: ArgumentError: New columns must have the same length as old columns
```
now errors, while:
```
julia> df = DataFrame(a=[])
0×1 DataFrame


julia> insertcols!(df, 1, :x => 9)
0×2 DataFrame
```
works just like:
```
julia> v = []
0-element Array{Any,1}

julia> v .= 9
0-element Array{Any,1}
```
works, but this fails:
```
julia> DataFrame(a=[], b=1)
ERROR: DimensionMismatch("column length 0 for column(s) a, and is incompatible with column length 1 for column(s) b")
```
So the question is if when we pseudo-broadcast a scalar into a 0-row container should that be allowed or no (I see pros and cons of both approaches and we are currently inconsistent already so we should pick one).

> `combine(gdf, :, :x1 => mean)`  works and "spreads" everything.

yes, it would spread (it is just not implemented yet)